FreeRTOS/FreeRTOS-Kernel
# Implemented heap_6 for C++ real-time applications

Description
-----------
I've adjusted my heap to FreeRTOS like posted in the forum:

https://forums.freertos.org/t/real-time-memory-manager/19685

This is my very first commit to FreeRTOS, i don't even know how to compile the kernel. Maybe You get it running and let me know what You think about it.


Test Steps
-----------
Because i don't know how to compile FreeRTOS, i did my tests in a simple console application.
I allocated variable sized blocks, deleted every second entry and allocated again.

You can find my test here: 
[HeapTest.zip](https://github.com/FreeRTOS/FreeRTOS-Kernel/files/14968482/HeapTest.zip)

By submitting this pull request, I confirm that You can use, modify, copy, and redistribute this contribution, under the terms of Your choice.

Best regards,

Sven Bieg
